uaclient/update_info.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Dict
+
+from uaclient import util
+from uaclient.config import UAConfig
+
+# We are doing this because we are sure that python3-apt comes with the distro,
+# but it cannot be installed in a virtual environment to be properly tested.
+# All our unit tests are mocking the Cache object, so we are not losing
+# anything here. If by any chance there is no python3-apt in the future, the
+# integration tests should be able to detect it through failure, because we
+# don't expect an empty apt cache there.
+try:
+    from apt import Cache  # type: ignore
+except ImportError:
+
+    def Cache():
+        return []
+
+
+ESM_SERVICES = ("esm-infra", "esm-apps")
+
+series = util.get_platform_info()["series"]
```

Comment:
This is not for this PR but just an observation that I made recently that this reminds me of:

We call `get_platform_info` all over the codebase, and I don't think the result should ever change during one run, but it reads the os release file every single time. For that reason, it is good that you avoid calling this repeatedly (though it makes unit testing harder I think, see below). I _think_ we could just throw a `lru_cache` on this function and then call it repeatedly without remorse - that would clean up our log files as well, which currently log that we're reading that os file all the time. I'm not sure if there might be any unintended consequences to that though, @lucasmoura @blackboxsw thoughts?